api/main.py

```diff
@@ -349,21 +454,49 @@ async def search(
     return JSONResponse(content=results)
 
 
-@app.put(
+@app.post(
     "/add_standards",
     response_class=HTMLResponse,
     dependencies=[Depends(RateLimiter(times=rate_times, seconds=rate_seconds))],
 )
 async def add_standards(request: Request, doc: dict):
     """Add standards to the main Elasticsearch index by PUTTING a JSON request here."""
-    # TODO: Check Standard body.
     res = es.index(index=idx_main, body=json.dumps(doc))
     print(res)
     json_compatible_item_data = jsonable_encoder(doc)
     log_stats(request, data=doc)
     return JSONResponse(content=json_compatible_item_data)
 
 
+@app.put(
+    "/edit_standards",
+    response_class=HTMLResponse,
+    dependencies=[Depends(RateLimiter(times=rate_times, seconds=rate_seconds))],
+)
+async def edit_standards(request: Request, id: str, doc: dict):
+    """Add standards to the main Elasticsearch index by PUTTING a JSON request here."""
+    res = es.update(index=idx_main, id=id, body=json.dumps(doc))
+    print(res)
+    json_compatible_item_data = jsonable_encoder(doc)
+    log_stats(request, data=doc)
+    return JSONResponse(content=json_compatible_item_data)
+
+
+@app.delete(
+    "/delete_standards",
+    response_class=HTMLResponse,
+    dependencies=[Depends(RateLimiter(times=rate_times, seconds=rate_seconds))],
+)
+async def delete_standards(request: Request, id: str, doc: dict):
+    """Delete standards to the main Elasticsearch index by PUTTING a JSON request here."""
+    # TODO: Once we are connected to LDAP, add line to verify auth of Admin.
+    res = es.delete(index=idx_main, id=id)
+    print(res)
+    json_compatible_item_data = jsonable_encoder(doc)
+    log_stats(request, data=doc)
+    return JSONResponse(content=json_compatible_item_data)
+
+
```

Comment:
@vlall these will also need to call the plugins for indexers and vector starage to make sure they are in-sync